### ISSUE TITLE
Rule S3353: False positive.

### ISSUE THREAD

**Issue by Danghor** [SonarSource/sonar-dotnet#2499]
### Description

False positive for _Rule S3353: Unchanged local variables should be "const"_ found.

![Capture](https://user-images.githubusercontent.com/7728268/61689260-2524a880-ad27-11e9-964a-163b68ec8fca.PNG)

### Repro steps

For the following code, [Rule S3353](https://rules.sonarsource.com/csharp/RSPEC-3353) detects `isValid` to be a `const` and suggests that I mark it as such:

```csharp
static readonly Func<string, bool> IsValidYearOfBirth = (year) =>
{
    bool isValid = false;

    if (short.TryParse(year, out short shortYear))
    {
        isValid = Prerequisites.IsValidYear(shortYear);
    }

    return isValid;
};
```

The sample is used [here](https://github.com/Danghor/Honglorn/blob/d6440299f785f1b946294063eb8fc43acd46f276/HonglornBL/Import/ImportedStudentRecord.cs#L68)

### Expected behavior

The rule should not be triggered.

### Actual behavior

The rule is triggered and suggests that `isValid` should be marked as `const`

### Known workarounds

The rule can be resolved as false positive.

### Related information

* SonarCloud for an open source project (AzureDevOps plugin "Run Code Analysis" 1.6.3)
* Visual Studio Version: 2017 Community (15.9.14)
* [SonarCloud](https://sonarcloud.io/project/issues?branch=develop&id=Honglorn&issues=AWwau2nMS8X08MRyfX0P&open=AWwau2nMS8X08MRyfX0P)
* [Azure DevOps](https://danghor.visualstudio.com/Honglorn/_build/results?buildId=184)
